Assets/Scripts/MeleeEnemy.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MeleeEnemy : MonoBehaviour
+{
+    public GameObject player;
+    [SerializeField] private float _speed;
+    [SerializeField] private float _attackCD;
+
+    private float _lastAttack;
+
+    private float _distance;
+    private PlayerHealth _playerHealth;
+
+    private void Awake()
+    {
+        _playerHealth = player.GetComponent<PlayerHealth>();
```

Comment:
Quite a taxing call to repeat over all enemies, maybe look into a Singleton to manage player health